pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/GetHexagonAddressFunction.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.geospatial.transform.function;
+
+import com.google.common.base.Preconditions;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.core.operator.blocks.ProjectionBlock;
+import org.apache.pinot.core.operator.transform.TransformResultMetadata;
+import org.apache.pinot.core.operator.transform.function.BaseTransformFunction;
+import org.apache.pinot.core.operator.transform.function.LiteralTransformFunction;
+import org.apache.pinot.core.operator.transform.function.TransformFunction;
+import org.apache.pinot.core.plan.DocIdSetPlanNode;
+import org.apache.pinot.segment.spi.datasource.DataSource;
+
+
+/**
+ *  A function that returns the H3 index address of a given geolocation.
+ */
+public class GetHexagonAddressFunction extends BaseTransformFunction {
+  public static final String FUNCTION_NAME = "GET_HEXAGON_ADDR";
+  private TransformFunction _firstArgument;
+  private TransformFunction _secondArgument;
+  private TransformFunction _thirdArgument;
+  private long[] _results;
+
+  @Override
+  public String getName() {
+    return FUNCTION_NAME;
+  }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, DataSource> dataSourceMap) {
+    Preconditions
```

Comment:
Should we also support 2 arguments version where the first argument is the serialized points?

Reply:
is it possible to have a different type of first ag? I thought func does not support polymorphism?

Reply:
Yes, we can handle different data type differently in transform function. Check `DistinctCountHLLAggregationFunction` as an example.

Reply:
updated